product_code_mandatory/models/product.py

```diff
@@ -0,0 +1,15 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators License AGPL-3.0
+# or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ProductProduct(models.Model):
```

Comment:
by default you create a product template, and you can add variants, the pre_init_hook is adding a default code to templates first, so here i think you should use the same, the product.template model.

Reply:
The pre_init_hook process in this module is to make sure that all existing records(product_product table) would have an initial value and not empty or blank. This is only done once upon installation of the module.